src/OmniSharp.Host/Endpoint/EndpointHandler.cs

```diff
@@ -142,18 +146,75 @@ private Task<object> HandleLanguageRequest(string language, TRequest request, Re
         {
             if (!string.IsNullOrEmpty(language))
             {
-                return HandleSingleRequest(language, request, packet);
+                return HandleRequestForLanguage(language, request, packet);
             }
 
             return HandleAllRequest(request, packet);
         }
 
-        private async Task<object> HandleSingleRequest(string language, TRequest request, RequestPacket packet)
+        private async Task<IAggregateResponse> AggregateResponsesFromLanguageHandlers(ExportHandler<TRequest, TResponse>[] handlers, TRequest request)
+        {
+            IAggregateResponse aggregateResponse = null;
+
+            var responses = new List<Task<TResponse>>();
+            foreach (var handler in handlers)
```

Comment:
we should probably short circuit here in case there is only 1? so that we don't create an unnecessary list or run Task.WhenAll

Reply:
Added special handling for the case when there is only a single handler for Endpoint/language combination. Interestingly enough I had to add a cast to IAggregateResponse while else case below doesn't need it.